tratto/src/main/java/star/tratto/util/javaparser/JDoctorUtils.java

```diff
@@ -0,0 +1,210 @@
+    ) {
+        return pathList.subList(0, pathList.size() - 1);
+    }
+
+    public static List<String> getPathList(
+            String path
+    ) {
+        String regex = "[.$]";
+        return Arrays.asList(path.split(regex));
+    }
+
+    public static String getJPTypeName(
+            TypeDeclaration<?> primaryType,
+            CallableDeclaration<?> jpCallable,
+            Parameter jpTypeParam
+    ) {
+        String jpTypeName = removeSpuriousCharacters(jpTypeParam.getType().asString());
+        JPType jpMethodType = JPType.getJPTypeDeclaration(jpTypeParam.getType());
+
+        if (jpMethodType == JPType.CLASS_OR_INTERFACE_TYPE) {
+            jpTypeName = removeSpuriousCharacters(jpTypeParam.getType().asClassOrInterfaceType().getName().asString());
+        }
+        if (hasJPTypeEllipsis(jpTypeParam.toString())) {
+            jpTypeName = convertToArrayType(jpTypeName, 1);
+        }
+        if (hasJPTypeExtends(jpCallable.getTokenRange().get().toString(), jpTypeName)) {
+            jpTypeName = getExtendsType(jpCallable.getTokenRange().get().toString(), jpTypeName);
+        }
+        if (primaryType.isClassOrInterfaceDeclaration()) {
+            ClassOrInterfaceDeclaration jpClass = primaryType.asClassOrInterfaceDeclaration();
+            if (hasJPClassTypeGenerics(jpClass)) {
+                for (TypeParameter jpClassType : jpClass.getTypeParameters()) {
+                    if (jpClassType.getNameAsString().equals(jpTypeName.replaceAll("\\[\\]",""))) {
+                        if (hasJPTypeExtends(jpClassType.toString(), jpTypeName)) {
+                            jpTypeName = getExtendsType(jpClassType.toString(), jpTypeName);
+                        }
+                    }
+                }
+            }
+        }
+        return jpTypeName;
+    }
+
+    public static boolean isStandardType(String conditionType) {
+        return conditionType.equals("Object") || conditionType.equals("Comparable");
+    }
+
+    public static boolean isStandardTypeArray(String conditionType) {
+        return conditionType.equals("Object[]") || conditionType.equals("Comparable[]");
+    }
+
+    public static boolean hasJPClassTypeGenerics(ClassOrInterfaceDeclaration jpClass) {
+        return jpClass.getTypeParameters().size() > 0;
+    }
+
+    public static boolean hasJPTypeEllipsis(String jpParamString) {
+        return jpParamString.contains("...");
+    }
+
+    public static boolean hasJPTypeExtends(String jpMethodString, String jpParamString) {
+        String regex = String.format("%s\\s+extends\\s+([A-Za-z0-9_]+)[<[A-Za-z0-9_,]+]*", jpParamString.replaceAll("\\[\\]",""));
+        // Using the Pattern and Matcher classes
+        Pattern pattern = Pattern.compile(regex);
+        Matcher matcher = pattern.matcher(jpMethodString);
+        return matcher.find();
+    }
+
+    public static String removeSpuriousCharacters(String dirtyType) {
+        String regex = "(super|\\s|\\?|;|<.*?>)";
```

Comment:
Should "super" be surrounded by "\b"?  Otherwise this will remove "super" from within a longer name.
Also, the .*? should forbid "<" (and ">" for symmetry, eliminating the need for the "?").